examples/plot_metric_examples.py

```diff
@@ -0,0 +1,410 @@
+"""
+Algorithms walkthrough
+~~~~~~~~~~~~~~~~~~~~~~
+
+This is a small walkthrough which illustrates all the Metric Learning
```

Comment:
not all actually

Reply:
Agreed, I will add MMC, but I don't know for MLKR, I think I'll just talk about it when describing NCA ? Since the cost function is very similar except that MLKR uses a soft-nearest neighbors for regression as I understood

Reply:
true, MLKR is for regression. it is a good idea to mention it there as proposed
maybe change "all the metric learning algorithms" to "most metric learning algorithms"

Reply:
In the end, I added most metric learning algorithms because we don't talk about `Covariance`, (we could maybe ?) And I added `MLKR` with `make_regression` task because I found the results were pretty cool too! Even if it breaks a bit the outline because it's just between supervised algorithms and the "constraints" section... Tell me what you think ?